Type fixes

DonovanDMC opened the pull request:
The `OldChannel` object, which is used in the [`channelUpdate`](https://github.com/abalabahaha/eris/blob/dev/index.d.ts#L665) event is missing the `nsfw`, and `type` properties, which are present in the [jsdoc](https://abal.moe/Eris/docs/Client#event-channelUpdate).

DonovanDMC committed:
OldChannel is missing nsfw & type

Skillz4Killz commented on index.d.ts:
```diff
@@ -74,7 +74,9 @@ declare namespace Eris {
     topic?: string;
+    type: 1 | 2 | 4 | 5 | 6;
     bitrate?: number;
     permissionOverwrites: Collection<PermissionOverwrite>;
```
```suggestion
    permissionOverwrites: Collection<PermissionOverwrite>;
    parentID?: string;
    rateLimitPerUser?: number;
```

Skillz4Killz reviewed (changes requested):
More missing types from oldChannel

DonovanDMC committed:
parentID & rateLimitPerUser

Co-Authored-By: Skillz4Killz <23035000+Skillz4Killz@users.noreply.github.com>

DonovanDMC committed:
type specifics for channels

DonovanDMC committed:
annoying but ok

DonovanDMC committed:
this being here breaks it

eritbh commented on index.d.ts:
```diff
@@ -662,7 +662,7 @@ declare namespace Eris {
       listener: (channel: GroupChannel, user: User) => void
     ): T;
-    (event: "channelUpdate", listener: (channel: AnyChannel, oldChannel: OldChannel) => void): T;
+    (event: "channelUpdate", listener: (channel: AnyChannel, oldChannel: GenericOldChannel & ({ type: 0; topic?: string; rateLimitPerUser: number } | { type: 2; bitrate: number } | { type: 5; topic?: string; rateLimitPerUser: 0 } | { type: 4 | 6 })) => void): T;
```
Wouldn't this be cleaner if we split it up into multiple named interfaces that all inherit from `GenericOldChannel`?

DonovanDMC replied:
We can do that, and add ~4-5 more interfaces, it would be cleaner, yes.

DonovanDMC committed:
Revert to simpler types & change guildRoleUpdate types.

DonovanDMC committed:
Make these not optional.

They were optional, due to the channel parameter being `AnyChannel`, but it's `AnyGuildChannel` now, therefore they will be there.

Skillz4Killz commented on index.d.ts:
```diff
@@ -74,9 +74,12 @@ declare namespace Eris {
+    parentID?: string;
+    topic?: string;
+    rateLimitPerUser?: number;
+    type: 0 | 2 | 4 | 5 | 6;
```
```suggestion
    type: 0 | 2 | 4 | 5 | 6;
    bitrate?: number;
```

Skillz4Killz commented on index.d.ts:
```diff
@@ -487,7 +490,19 @@ declare namespace Eris {
     color?: number;
     hoist?: boolean;
     mentionable?: boolean;
+    managed?: boolean;
```
I don't believe this is a valid option to pass.

DonovanDMC replied:
Not a valid option? It might not be able to be changed, but it is definitely sent
![image](https://user-images.githubusercontent.com/17226394/70070346-12658e80-15b9-11ea-9a07-a97e36109d6f.png)

DonovanDMC replied:
Oh, I see what you mean, I'm not 100% sure, let me go test things.

Skillz4Killz replied:
What you are editing is RoleOptions which is used for creating roles. Bots can't create managed roles so passing a managed thing is not a thing.

DonovanDMC replied:
Doesn't make a difference, so basically isn't a valid option
![image](https://user-images.githubusercontent.com/17226394/70070988-2b227400-15ba-11ea-8b51-fa52c68f8d69.png)

Skillz4Killz commented on index.d.ts:
```diff
@@ -662,7 +677,7 @@ declare namespace Eris {
       listener: (channel: GroupChannel, user: User) => void
     ): T;
-    (event: "channelUpdate", listener: (channel: AnyChannel, oldChannel: OldChannel) => void): T;
+    (event: "channelUpdate", listener: (channel: AnyGuildChannel, oldChannel: OldChannel) => void): T;
```
Although I agree, with this, it would basically be a breaking change for any GroupChannel bots.

DonovanDMC replied:
@Skillz4Killz to make that work, OldChannel would have to be changed a lot, it is completely wrong for Group DM bots.

Skillz4Killz replied:
I agree I don't think it is worth it to support GroupChannels so I agree this should be changed. I just think when merging this should be labelled as breaking.

DonovanDMC committed:
add bitrate

Co-Authored-By: Skillz4Killz <23035000+Skillz4Killz@users.noreply.github.com>

DonovanDMC committed:
Remove `managed` from RoleOptions.

DonovanDMC committed:
Merge branch 'dev' into patch-9

DonovanDMC committed:
Move `guildUpdate` from `GuildOptions` to a new `OldGuild`.

DonovanDMC committed:
forgot to replace with semicolons

DonovanDMC committed:
missed a semicolon

DonovanDMC committed:
`emojis` seems to be a bit different than originally thought.

DonovanDMC committed:
Same ol' semicolon stuff

abalabahaha committed:
Update index.d.ts

abalabahaha committed:
Merge branch 'dev' into patch-9
